### PR TITLE
[ROCm] Fix run single gpu script to avoid duplicates

### PR DESCRIPTION
**Pull request by Ruturaj4**
@superbobry can you please review and merge? thanks!

The script will help improve our internal reporting.

### CODE REVIEW

File: build/rocm/run_single_gpu.py [hunk 1]
```diff
@@ -115,7 +161,7 @@ def run_test(testmodule, gpu_tokens, continue_on_fail):
 
 
 def run_parallel(all_testmodules, p, c):
-  print(f"Running tests with parallelism=", p)
+  print(f"Running tests with parallelism = ", p)
```
Comment (superbobry) [inline build/rocm/run_single_gpu.py:164]: I think you might want to move `p` into the f-string?

File: build/rocm/run_single_gpu.py [hunk 2]
```diff
@@ -41,12 +70,18 @@ def generate_final_report(shell=False, env_vars={}):
...
+  # Generate json reports.
+  combine_json_reports()
+  # Generate csv reports.
+  combine_csv_reports()
 
 
 def run_shell_command(cmd, shell=False, env_vars={}):
-  env = os.environ
-  env = {**env, **env_vars}
+  # Create a copy of the current environment.
```
Comment (superbobry) [inline build/rocm/run_single_gpu.py:80]: FYI you can use unpacking for this

```
env = {**os.environ, **env_vars}
```

**Review by superbobry — APPROVED** [review]

File: build/rocm/run_single_gpu.py (hunk 1 above)
Comment (Ruturaj4) [reply build/rocm/run_single_gpu.py]: Thanks!

File: build/rocm/run_single_gpu.py (hunk 2 above)
Comment (Ruturaj4) [reply build/rocm/run_single_gpu.py]: Sure. Thanks!

**Review by superbobry — APPROVED** [review]

**Comment by Ruturaj4** [comment]
@superbobry is this not merged yet? just closed? Could you please check?

**Comment by dfm** [comment]
@Ruturaj4 — It looks like you closed this PR. Did you not mean to?

**Comment by Ruturaj4** [comment]
@dfm Just reopened again, idk how that was closed!

**Comment by dfm** [comment]
Thanks - can you rebase onto `main`? It looks like there are conflicts now.

**Comment by Ruturaj4** [comment]
> Thanks - can you rebase onto `main`? It looks like there are conflicts now.

Done, thanks!

**Comment by dfm** [comment]
@Ruturaj4 — Thanks! Please also squash the commits into one and we'll get this merged.

**Comment by Ruturaj4** [comment]
@dfm thanks! I will keep in mind two squash commits in future. can you please approve?

**Review by dfm — APPROVED** [review]

**Pull request merged**